examples/meta/src/evaluation/cross_validation_pipeline.sg

```diff
@@ -0,0 +1,33 @@
+File f_feats_train = csv_file("../../data/classifier_binary_2d_linear_features_train.dat")
+File f_feats_test = csv_file("../../data/classifier_binary_2d_linear_features_test.dat")
+File f_labels_train = csv_file("../../data/classifier_binary_2d_linear_labels_train.dat")
+File f_labels_test = csv_file("../../data/classifier_binary_2d_linear_labels_test.dat")
+
+#![create_features]
+Features feats_train = features(f_feats_train)
+Features feats_test = features(f_feats_test)
+Labels labels_train = labels(f_labels_train)
+Labels labels_test = labels(f_labels_test)
+#![create_features]
+
+#![create_pipeline
+Transformer subMean = transformer("PruneVarSubMean")
+Machine svm = machine("LibLinear")
+
+PipelineBuilder builder()
+builder.over(subMean)
+Pipeline pipeline = builder.then(svm)
+#![create_pipeline]
+
+#![create_cross_validation]
+StratifiedCrossValidationSplitting splitting_strategy(labels_train, 2)
```

Comment:
dont we have factories for those things?
I would prefer if all newly added examples would fully use the new api so we dont have to refactor later

Reply:
we don't have factory for `CSplittingStrategy`

Reply:
Would you mind creating one? That should be pretty easy!

Reply:
sure

Reply:
@karlnapf since there are multiple subclasses of `CSplittingStrategy`, are we going to do some string comparison by name?

Reply:
check the factory `machine` in `factory.h, you just have to add some macro lines
the call should be `splitting_strategy("StratifiedCrossValidationSplitting", labels=labels_train, k=2)`

Reply:
so we need to do initialization in `init()` method, instead of the constructor
e.g. https://github.com/shogun-toolbox/shogun/blob/1be1dbbf6fdce20e73901d205cee253f2d4cd873/src/shogun/evaluation/StratifiedCrossValidationSplitting.cpp#L21
this need refactor right?

Reply:
ah yes, this will be moved into the method that is called from outside, i.e. `build_subsets` ... putting it into a helper method makes sense